packages/astro/src/runtime/server/render/server-islands.ts

```diff
@@ -105,8 +105,31 @@ export function renderServerIsland(
 				);
 			}
 
-			destination.write(`<script async type="module" data-astro-rerun data-island-id="${hostId}">
-let script = document.querySelector('script[data-island-id="${hostId}"]');
+			if(!result._metadata.hasServerIslandScript) {
+				result._metadata.hasServerIslandScript = true;
+// The runtime is a non-module script so that we can guarantee that is has executed before any module scripts
+// This is important because the module script will call replaceServerIsland, which needs to be defined first
+// It's ok to make it a sync script because it doesn't do anything until the module script runs
+				destination.write(`<script data-si-runtime>
```

Comment:
if this is only used for the test, it would be a shame to always have this in peoples builds. Can you test by getting the script contents and looking for some expected content?